hopsworks-web/yo/app/scripts/controllers/featuregroupViewInfoCtrl.js

```diff
@@ -201,70 +196,136 @@ angular.module('hopsWorksApp')
+                }
+
+                self.selectedFeaturegroup = featuregroups.versionToGroups[featuregroups.activeVersion]
+
+                self.projectId = projectId;
+                self.projectName = projectName;
+                self.featurestore = featurestore;
+                self.featuregroups = featuregroups;
+                self.activeVersion = featuregroups.activeVersion;
+                self.settings = settings;
+
+                self.cachedFeaturegroupType = self.settings.cachedFeaturegroupType;
+                self.onDemandFeaturegroupType = self.settings.onDemandFeaturegroupType;
+                self.pythonCode = self.getPythonCode();
+                self.scalaCode = self.getScalaCode();
+
+                self.featuregroupType = "";
+                if(self.selectedFeaturegroup.featuregroupType === self.onDemandFeaturegroupType){
+                    self.featuregroupType = "ON DEMAND";
                 } else {
                     self.featuregroupType = "CACHED";
                     self.fetchSchema();
                     self.fetchSize();
                     self.fetchSample();
                     self.fetchCustomMetadata();
                 }
+
             };
 
             /**
-             * Format javascript date as string (YYYY-mm-dd HH:MM:SS)
+             * Helper function for redirecting to another project page
              *
-             * @param d date to format
-             * @returns {string} formatted string
+             * @param serviceName project page
              */
-            self.formatDate = function(javaDate) {
-                var d = new Date(javaDate)
-                var date_format_str = d.getFullYear().toString()+"-"+((d.getMonth()+1).toString().length==2?(d.getMonth()+1).toString():"0"+(d.getMonth()+1).toString())+"-"+(d.getDate().toString().length==2?d.getDate().toString():"0"+d.getDate().toString())+" "+(d.getHours().toString().length==2?d.getHours().toString():"0"+d.getHours().toString())+":"+((parseInt(d.getMinutes()/5)*5).toString().length==2?(parseInt(d.getMinutes()/5)*5).toString():"0"+(parseInt(d.getMinutes()/5)*5).toString())+":00";
-                return date_format_str
-            }
+            self.goToUrl = function (serviceName) {
+                $location.path('project/' + self.projectId + '/' + serviceName);
+            };
+
+            self.goToDataValidation = function () {
+                StorageService.store("dv_featuregroup", self.selectedFeaturegroup);
+                $location.path('project/' + self.projectId + "/featurestore/datavalidation");
+            };
 
             /**
-             * Closes the modal
+             * Called when the increment-version-featuregroup-button is pressed
+             *
              */
-            self.close = function () {
-                $uibModalInstance.dismiss('cancel');
+            self.newFeaturegroupVersion = function () {
+                var i;
```

Comment:
Truth to be told I'm not sure what's the current behavior. But here I'd expect to see `selectedFeaturegroup` instead of finding the max version. Meaning that users can create "a new version" from version 3 when the newest version is 5.

Reply:
I agree, I assumed that's how it should be working too but does not look like it on master: https://github.com/logicalclocks/hopsworks/blob/9ba01166dc7b61f9fa7069278449cf191b672304/hopsworks-web/yo/app/scripts/controllers/featurestoreCtrl.js#L532 I can fix it.